src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -44,17 +44,19 @@ var (
 	errNonexistentLastObservesState = errors.New("expecting to have statefulset LastObservedState set but it's nil")
 	errNodePortMissing              = errors.New("the node port is missing from the service")
 	errInvalidImagePullPolicy       = errors.New("invalid image pull policy")
+	ErrRestrictedVersion            = errors.New("cluster management restricted by version")
```

Comment:
Have you considered the option to just logging and return without error? We don't need logging errors and the associated exponential backoff as the cluster will be again reconciled after we change its version externally.

Also, this feature can be technically used to run multiple operators on the same cluster (I did it in the past in a managed service), managing different `Cluster` resource. In that case, both operators will throw a lot of error logs under normal working conditions.

Reply:
I removed the error and adjusted the test to due a "consistently" test for 1 second.

Reply:
Sorry, I missed the notification. Sounds good